homeassistant/components/sensor/recollect_waste.py

```diff
@@ -0,0 +1,100 @@
+"""
+Support for Recollect Waste curbside collection pickup.
+
+For more details about this platform, please refer to the documentation at
+https://www.home-assistant.io/components/sensor.recollect_waste/
+"""
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+import homeassistant.helpers.config_validation as cv
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+from homeassistant.const import (CONF_NAME, CONF_SCAN_INTERVAL)
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['recollect-waste==1.0.1']
+
+_LOGGER = logging.getLogger(__name__)
+ATTR_PICKUP_TYPES = 'pickup_types'
+ATTR_AREA_NAME = 'area_name'
+CONF_PLACE_ID = 'place_id'
+CONF_SERVICE_ID = 'service_id'
+DEFAULT_SCAN_INTERVAL = timedelta(days=1)
+DOMAIN = 'recollect_waste'
+ICON = 'mdi:trash-can-outline'
+
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_PLACE_ID): cv.string,
+    vol.Required(CONF_SERVICE_ID): cv.string,
+    vol.Optional(CONF_NAME, default=DOMAIN): cv.string,
+    vol.Optional(CONF_SCAN_INTERVAL, default=DEFAULT_SCAN_INTERVAL): (
+        vol.All(cv.time_period, cv.positive_timedelta))
+})
+
+
+def setup_platform(hass, config, add_entities, discovery_info=None):
+    """Set up the Recollect Waste platform."""
+    add_entities([RecollectWasteSensor(
+        config.get(CONF_NAME),
+        config[CONF_PLACE_ID],
+        config[CONF_SERVICE_ID],
+        config.get(CONF_SCAN_INTERVAL))], True)
+
+
+class RecollectWasteSensor(Entity):
+    """Recollect Waste Sensor."""
+
+    def __init__(self, name, place_id, service_id, scan_interval):
+        """Initialize the sensor."""
+        self._attributes = {}
+        self._name = name
+        self._state = None
+        self.place_id = place_id
+        self.service_id = service_id
+        self.update = Throttle(scan_interval)(self._update)
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self._name
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        return "{}{}".format(self.place_id, self.service_id)
```

Comment:
I was asked by @bachya to add the unique_id. Maybe I misunderstood how to add it properly as the documentation is sparse on it.  https://github.com/home-assistant/home-assistant/pull/20121#discussion_r247907771

Reply:
Yes. What do the place id and service id represent?

Reply:
They are two properties that define the parameters used to retrieve the waste pickup information from the API URL. I figured it would be a good unique identifier since I cannot see anyone setting up multiple sensors with the same place and service ids. Maybe there is a better way? Should I keep this unique_id property here still?

Reply:
I think it sounds ok like this. It's similar to sensor placement coordinate that we have allowed before.